drivers/soundwire/intel.c

```diff
@@ -1739,20 +1741,31 @@ static int intel_resume_runtime(struct device *dev)
 			return ret;
 		}
 
+		/*
+		 * There is a exception condition for CLK_STOP_BUS_RESET
+		 * case. If there is another master is working, all the masters
+		 * are powered on for they are in the same power domain.
+		 * Master can preserve its context for clock stop0, so there is
+		 * no need to clear slave status and reset bus.
+		 */
+		clock_stop0 = sdw_cdns_is_clock_stop(&sdw->cdns);
+
```

Comment:
@RanderWang @bardliao I am not sure I get the idea.
Is it the case that when the power is removed, the IP will be reset and will not show as being in clock stop mode any longer?

Reply:
@plbossart  yes, it is exactly what you said.   if the master is in clock stop0, it means master preserve its context, so clock stop0 can work, or it is what you said

Reply:
ok, can you rebase and update comments and I'll merge.

I think what you meant is this:

When all the links are suspended, the HDaudio controller may suspend and the power rails to the SoundWire IP may be disabled, requiring a complete re-initialization/enumeration on resume.
However, if one or more Masters remained active, the HDaudio controller will remain active and the power rails will remain enabled.
As a result, during the link resume step we can check if the context was preserved  by verifying if the clock was stopped, and avoid doing a complete bus reset and re-enumeration.

Reply:
yes, I am working now

Reply:
btw @RanderWang, if I can give a piece of advice: avoid using 'there is' in commit message or bug reports. It's always difficult to figure out what 'there' refers to. Use an explicit subject to make sure the reader knows exactly what you are referring to. Also avoid 'work' and use concepts that are related to the subsystem (for pm_runtime it's active and suspended).

e.g.

There is a exception condition for CLK_STOP_BUS_RESET
case if there is another master is working

->

An exception condition occurs for the CLK_STOP_BUS_RESET case if one or more masters remain active.

Reply:
@plbossart  thanks very much,  I update my comments.